app/js/scripts.js

```diff
@@ -134,148 +134,239 @@ function editReportById(reportId){
     })
     $('#fields-to-select').selectpicker('val', fieldsToSelectArr);
     $("#fields-to-select").selectpicker("refresh");
+    console.log(reportObject.show_in_dashboard);
     //Show the new report UI
     changeReportView('computer', 'edit');
+    $("#show-in-dashboard").prop('checked', reportObject.show_in_dashboard);
     $("#report-name-field").html(reportObject.name);
     $("#new-report-name").val(reportObject.name);
     $("#existing-report-id").val(reportId)
-    updateQueryStringParam('reportId',reportId);
+    updateQueryStringParam('reportId', reportId);
   })
-  .fail(function(xhr){
-    console.log(xhr);
-  });
+    .fail(function (xhr) {
+      console.log(xhr);
+    });
 }
 
-function runExistingReportByUrl(){
+function loadReportById(reportId) {
+  reloadReportPane(false);
+  //Make the request to the server to get a saved report
+  var report = getRequestObject('/reports/id/' + reportId, null, 'GET');
+  report.done(function (reportObject) {
+    //Load the existing report view
+    $("#new-report-parent").hide();
+    console.log(reportObject.line_items);
+    addMultipleReportLineItems(reportObject.line_items);
+    //Fill in and show the fields to select
+    //Show the new report UI
+    changeReportView('computer', 'view');
+    $("#report-name-field").html(reportObject.name);
+    updateQueryStringParam('reportId', reportId);
+  })
+    .fail(function (xhr) {
+      console.log(xhr);
+    })
+}
+
+async function calculateReportResultCount() {
+  var reportObject = await asyncGetRequestObject('/reports/dashboard/', null, 'GET'); //gets all reports in mySQL
+  var outputObj = reportObject;
+  var reportCalls = [];
+  for (var i = 0; i < reportObject.length; i++) { //for each report
+    reportCalls = await asyncGetRequestObject('/reports/' + reportObject[i].id + '/count', null, 'GET');
```

Comment:
You could consider using Promise.all() here to kick these off all at the same time and not block in the loop